state/protocol/badger/mutator.go

```diff
@@ -104,8 +104,15 @@ func NewFullConsensusState(
 // Instead, the consensus follower relies on the consensus participants to
 // validate the full payload. Therefore, a follower a QC (i.e. a child block) as
 // proof that a block is valid.
+// Expected errors during normal operations:
+//   - state.OutdatedExtensionError if the candidate block is outdated (e.g. orphaned)
+//   - state.InvalidExtensionError if the candidate block is invalid
 func (m *FollowerState) Extend(ctx context.Context, candidate *flow.Block) error {
 
+	if err := candidate.StructureValid(); err != nil {
```

Comment:
## ❓ 

I am confused why we are calling `StructureValid` here again. Didn't the networking layer already check this. 
I would really prefer a design, where is very clear where certain checks happen. Otherwise, I think we amplify the risk of checking some things 10 times and still miss some other necessary checks.